libminifi/test/unit/ThreadPoolTests.cpp

```diff
@@ -84,7 +81,33 @@ TEST_CASE("ThreadPoolTest2", "[TPT2]") {
   utils::Worker<int> functor(f_ex, "id", std::move(after_execute));
   pool.start();
   std::future<int> fut;
-  pool.execute(std::move(functor), fut);  // NOLINT(bugprone-use-after-move)
+  pool.execute(std::move(functor), fut);
   fut.wait();
   REQUIRE(20 == fut.get());
 }
+
+TEST_CASE("Worker wait time should be relative to the last run") {
+  std::mutex cv_m;
+  std::condition_variable cv;
```

Comment:
You could use `std::promise<void>` or `std::atomic_flag` (like #1576) for a simpler code.

Reply:
I've simplified this test in https://github.com/apache/nifi-minifi-cpp/pull/1581/commits/187f07eb04efef01712e81cd9f3fc1dfa93bc93d